### PR TITLE
creating form for new todos

### PR DESCRIPTION
**Pull request by 1stIndominus**
[DEMO LINK](https://1stIndominus.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by qoa11a — CHANGES_REQUESTED** [review]
image
There's still error if I didn't select user first time and then have selected it.

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,118 @@
...
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App:React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState([...todos]);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [errorExistence, setErrorExistence] = useState(false);
+  const [selectError, setSelectError] = useState(false);
```
Comment (liliia-livandovskaya) [inline src/App.tsx:26]:
```suggestion
  const [isSelectedValid, setSelectError] = useState(false);
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,118 @@
...
+          User:&nbsp;
+          <select
+            data-cy="userSelect"
+            name="todoUser"
+            value={selectedUser}
+            onChange={handleUserName}
+          >
+            <option value={0} disabled>Choose a user</option>
+            {usersFromServer.map(user => (
+              <option key={user.id} value={user.id}>{user.name}</option>
```
Comment (liliia-livandovskaya) [inline src/App.tsx:101]:
```suggestion
              <option key={user.id} value={user.id}>
                {user.name}
              </option>
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,118 @@
...
+
+  const handleUserName = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(+event.target.value);
+    setErrorExistence(false);
+  };
+
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim().length) {
```
Comment (liliia-livandovskaya) [inline src/App.tsx:41]: let's create a variable for` !title.trim().length` and use it there

**Review by liliia-livandovskaya — CHANGES_REQUESTED** [review]
Great job! Pay attention to the comments

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,122 @@
...
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App:React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState([...todos]);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [errorExistence, setErrorExistence] = useState(false);
+  const [selectError, setSelectError] = useState(false);
```
Comment (polosanya) [inline src/App.tsx:26]: image
Try to rename your boolean variables according to this rule

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,122 @@
...
+
+  const titleTrim = title.trim().length;
```
Comment (polosanya) [inline src/App.tsx:28]: You can create it inside handleSubmit function, as you don't use it anywhere else

File: src/App.tsx [hunk 6]
```diff
@@ -1,61 +1,122 @@
...
+            {usersFromServer.map(user => (
+              <option key={user.id} value={user.id}>
+                {user.name}
+              </option>
+            ))}
           </select>
 
-          <span className="error">Please choose a user</span>
+          {errorExistence && (
+            <span className="error">Plese choose a user</span>
```
Comment (polosanya) [inline src/App.tsx:110]:
```suggestion
            <span className="error">Please choose a user</span>
```
Probably that's the reason why you can't pass the test

**Review by polosanya — CHANGES_REQUESTED** [review]
Almost done! 
Please pay attention to my comments

**Review by liliia-livandovskaya — APPROVED** [review]
Great job 🔥

**Pull request closed without merging**